src/main/java/sirius/biz/storage/layer2/BlobDispatcher.java

```diff
@@ -384,19 +388,37 @@ private void virtualDelivery(WebContext request,
                                  String blobKey,
                                  String filename,
                                  boolean download,
-                                 boolean cacheable) {
+                                 boolean cacheable,
+                                 boolean largeFileExpected) {
         if (!utils.verifyHash(Strings.isFilled(variant) ? blobKey + "-" + variant : blobKey, accessToken)) {
             request.respondWith().error(HttpResponseStatus.UNAUTHORIZED);
             return;
         }
 
         BlobStorageSpace storageSpace = blobStorage.getSpace(space);
         Response response = request.respondWith();
+
         if (cacheable) {
-            response.cached();
+            // If a virtual request is marked as cacheable, we try to redirect to the proper physical blob key
+            // as this will remain in cache much longer (and the redirect itself will also be cached). The additional
+            // HTTP round-trip for the redirect shouldn't hurt too much, as it is most probably optimized away due to
+            // keep-alive. However, using a physical delivery with infinite cache settings will enable any downstream
+            // reverse-proxies to maximize their cache utilization...
+            if (attemptRedirectToPhysical(response,
```

Comment:
Just a design topic: I don't like the approach of changing the response inside the method (makes reading it more complex). Would make the method return the url, and if non empty set the response/redirect hier.

Reply:
In general I don't like the pattern of altering a parameter inside a method, too. But in the case of the response here it is not really "altering" the response but completing it?
But I also agree that keeping all actually response calls in this method and not extract them away would simplify understanding what can be the response we send out.